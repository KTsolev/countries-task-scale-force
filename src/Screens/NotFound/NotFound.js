import React from 'react';
import { useHistory } from "react-router-dom";
import {ReactComponent as BackArrow} from '../../assets/back_arrow_left.svg';
import NotFound from '../../assets/not-found-img.jpg';

export const NotFoundPage = (props) => {
    const history = useHistory();

    const goBack = () => history.replace('/');

    return (
        <section className='not-found'>
            <div className='not-found-item' onClick={goBack}>
                <BackArrow className='svg' />
            </div>
            <div className='not-found-img'>
                <img src={NotFound} alt='not found page' />
            </div>
        </section>  
    )
}