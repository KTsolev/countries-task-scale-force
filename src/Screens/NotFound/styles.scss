.not-found {
    @include column;
    justify-content: center;
    align-items: center;
    position: relative;

    &-item {
        @include row;
        justify-content: space-around;
        align-items: stretch;

        .svg {
            position: absolute;
            left: 10px;
            top: 10px;
            padding: 2px 4px;
            width: 30px;
            height: 30px;
        }
    }

    &-img {
        @include row;
        justify-content: center;
        align-items: center;
        flex: 1;
        width: 100%;
        height: 80vh;
        min-height: 500px;
        min-width:300px;

        img {
            width: 100%;
            height: 100%;
        }
    }
}