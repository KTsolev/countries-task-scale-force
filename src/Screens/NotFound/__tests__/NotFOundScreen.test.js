import React from 'react';
import { NotFoundPage } from '../NotFound';
import renderer from 'react-test-renderer'; 

const mockedHistory = jest.fn();

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: jest.fn(),
      replace: mockedHistory,
    }),
  }));

describe('NotFoundPage tests', () => {
    it('should render component without props', () => {
       const tree = renderer.create(<NotFoundPage />);
        
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should handle on click', () => {
        const tree = renderer.create(<NotFoundPage />).root;
         
        const button = tree.findByProps({className: 'not-found-item'});

        expect(button).toBeTruthy();

        button.props.onClick();

        expect(mockedHistory).toHaveBeenCalled();
     });
});