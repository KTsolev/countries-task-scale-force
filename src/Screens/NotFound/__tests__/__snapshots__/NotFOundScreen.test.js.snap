// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage tests should render component without props 1`] = `
<section
  className="not-found"
>
  <div
    className="not-found-item"
    onClick={[Function]}
  >
    <test
      className="svg"
    />
  </div>
  <div
    className="not-found-img"
  >
    <img
      alt="not found page"
      src="test"
    />
  </div>
</section>
`;
